kotlin/flatbuffers-kotlin/src/commonMain/kotlin/com/google/flatbuffers/kotlin/JSON.kt

```diff
@@ -0,0 +1,788 @@
+      currentChar = fetch(cursorPos)
+      if (!isPlainStringChar(currentChar)) {
+        foundEscape = true
+        break
+      }
+      currentChar = fetch(cursorPos + 1)
+      if (!isPlainStringChar(currentChar)) {
+        cursorPos += 1
+        foundEscape = true
+        break
+      }
+      currentChar = fetch(cursorPos + 2)
+      if (!isPlainStringChar(currentChar)) {
+        cursorPos += 2
+        foundEscape = true
+        break
+      }
+      currentChar = fetch(cursorPos + 3)
+      if (!isPlainStringChar(currentChar)) {
+        cursorPos += 3
+        foundEscape = true
+        break
+      }
+      cursorPos += 4
+    }
+    if (!foundEscape) {
+      // if non-plain string char is not found we loop over
+      // the remaining bytes
+      while (true) {
+        if (cursorPos >= limit) {
+          error("Unexpected end of string")
+        }
+        currentChar = fetch(cursorPos)
+        if (!isPlainStringChar(currentChar)) {
+          break
+        }
+        ++cursorPos
+      }
+    }
+    if (currentChar == LETTER_DOUBLE_QUOTE) {
+      val str = data.getString(readPos, cursorPos - readPos)
+      readPos = cursorPos + 1
+      return str
+    }
+    if (currentChar in 0..0x1f) {
+      error("Illegal Codepoint")
+    } else {
+      // backslash or >0x7f
+      return readStringSlow(data, currentChar, cursorPos)
+    }
+  }
+
+  private fun readStringSlow(data: ReadBuffer, first: Byte, lastPos: Int): String {
+    var cursorPos = lastPos
+
+    var endOfString = lastPos
+    while (true) {
+      val pos = data.findFirst(LETTER_DOUBLE_QUOTE, endOfString)
+      when {
+        pos == -1 -> makeError(data, "Unexpected EOF, missing end of string'\"'", first)
+        data[pos - 1] == LETTER_BACKSLASH -> endOfString = pos + 1
+        else -> {
+          endOfString = pos; break
+        }
+      }
+    }
+    // copy everything before the escape
+    val builder = StringBuilder(data.getString(readPos, lastPos - readPos))
+    while (true) {
+      when (val pos = data.findFirst(LETTER_BACKSLASH, cursorPos, endOfString)) {
+        -1 -> {
+          val doubleQuotePos = data.findFirst(LETTER_DOUBLE_QUOTE, endOfString)
+          if (doubleQuotePos == -1) makeError(data, "Reached EOF before enclosing string", first)
+          val rest = data.getString(cursorPos, doubleQuotePos - cursorPos)
+          builder.append(rest)
+          readPos = doubleQuotePos + 1
+          return builder.toString()
+        }
+
+        else -> {
+          // we write everything up to \
+          builder.append(data.getString(cursorPos, pos - cursorPos))
+          val c = data[pos + 1]
+          builder.append(readEscapedChar(data, c, pos))
+          cursorPos = pos + if (c == LETTER_u) 6 else 2
+        }
+      }
+    }
+  }
+
+  // byte1 is expected to be first char before `\`
+  private inline fun readEscapedChar(data: ReadBuffer, byte1: Byte, cursorPos: Int): Char {
+    return when (byte1) {
```

Comment:
would be good to check this string/escape parsing code against what is in `idl_parser.cpp` for this, as that is generally what FlatBuffers accepts in terms of "json" :)